### PR TITLE
Feature/uar 766 error scenario at least one active mo

### PR DESCRIPTION
**Pull request by cdrainCH**
### JIRA link

https://companieshouse.atlassian.net/jira/software/c/projects/UAR/boards/531?selectedIssue=UAR-766&sprint=2999

### Change description

Adds an error scenario for when the user selects the 'All beneficial owners have been identified and I can provide all the required information' statement but there is an active managing officer.

Screenshot 2023-08-24 at 11 04 47

### Work checklist

- [ ] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/utils/application.data.ts [hunk 1]
```diff
@@ -77,14 +77,14 @@ export const checkActiveBOExists = (appData: ApplicationData): boolean => {
...
-};
-
 export const checkActiveMOExists = (appData: ApplicationData): boolean => {
   return allManagingOfficers(appData).some((mo) => !mo.resigned_on || Object.keys(mo.resigned_on).length === 0);
 };
 
+export const checkBOsDetailsEntered = (appData: ApplicationData): boolean => {
+  return Boolean( appData[BeneficialOwnerIndividualKey]?.length || appData[BeneficialOwnerOtherKey]?.length || appData[BeneficialOwnerGovKey]?.length );
+};
+
```
Comment (corybudd-ch) [inline src/utils/application.data.ts:87]: is this change needed?

File: src/middleware/statement.validation.middleware.ts [hunk 2]
```diff
@@ -29,6 +29,8 @@ export const hasValidStatements = (req: Request, res: Response, next: NextFuncti
 
 const validateIdentifiedBOsStatement = (appData: ApplicationData, errorList: string[]): boolean => {
   const allOrSomeBOsIdentified: boolean = (appData[BeneficialOwnerStatementKey] === BeneficialOwnersStatementType.ALL_IDENTIFIED_ALL_DETAILS || appData[BeneficialOwnerStatementKey] === BeneficialOwnersStatementType.SOME_IDENTIFIED_ALL_DETAILS);
+  const someOrNoneBOsIdentified: boolean = (appData[BeneficialOwnerStatementKey] === BeneficialOwnersStatementType.SOME_IDENTIFIED_ALL_DETAILS || appData[BeneficialOwnerStatementKey] === BeneficialOwnersStatementType.NONE_IDENTIFIED);
+  const allBOs: boolean = (appData[BeneficialOwnerStatementKey] === BeneficialOwnersStatementType.ALL_IDENTIFIED_ALL_DETAILS);
```
Comment (corybudd-ch) [inline src/middleware/statement.validation.middleware.ts:33]: Call it `allBOsIdentified` so it is statement related

File: test/controllers/update/update.statement.validation.errors.controller.spec.ts [hunk 3]
```diff
@@ -177,6 +178,72 @@ describe("Update statement validation errors controller", () => {
...
+      expect(resp.text).toContain('There is at least one active managing officer.');
+      expect(resp.text).toContain('All beneficial owners have been identified and I can provide all the required information');
+      expect(resp.text).toContain('The entity has no reasonable cause to believe that anyone has become or ceased to be a registrable beneficial owner during the update period');
+
+      expect(resp.text).not.toContain(UPDATE_REGISTRABLE_BENEFICIAL_OWNER_URL);
+      expect(resp.text).not.toContain('value="statement-resolution-change-information" checked');
+      expect(resp.text).not.toContain('value="statement-resolution-change-statement" checked');
+      expect(resp.text).not.toContain(PAGE_TITLE_ERROR);
+    });
+
```
Comment (corybudd-ch) [inline test/controllers/update/update.statement.validation.errors.controller.spec.ts:246]: this test file keeps growing and it is a lot of repeated checks - are these tests really necessary compared to ones that are already there? I don't think we need to have separate tests in here for every scenario because your tests in _test/middleware/statement.validation.middleware.spec.ts_ cover all the error scenarios.

You could just add your error message into `req['statementErrorList']` of an existing test but even then not sure what value you're getting out of that.

File: test/utils/application.data.spec.ts [hunk 4]
```diff
@@ -473,9 +473,9 @@ describe("Application data utils", () => {
       MANAGING_OFFICER_CORPORATE_OBJECT_MOCK
     ]
   ])(`allManagingOfficers with %s returns expected array`, (_, moKey, moMock) => {
-    const allMOs = allManagingOfficers({ [UpdateKey]: { [moKey]: [moMock] } });
+    const allBOs = allManagingOfficers({ [UpdateKey]: { [moKey]: [moMock] } });
 
-    expect(allMOs).toEqual([moMock]);
+    expect(allBOs).toEqual([moMock]);
```
Comment (corybudd-ch) [inline test/utils/application.data.spec.ts:478]: think you maybe need to pull the latest from main for this file, I had these called BOs but changed it to MOs

File: test/controllers/update/update.statement.validation.errors.controller.spec.ts (hunk 3 above)
Comment (corybudd-ch) [reply test/controllers/update/update.statement.validation.errors.controller.spec.ts:246]: yeah sweet go for it 💯 ! Don't think we'll lose anything without it

File: test/utils/application.data.spec.ts (hunk 4 above)
Comment (cdrainCH) [reply test/utils/application.data.spec.ts]: Fixed

**Review by corybudd-ch — APPROVED** [review]

**Review by joshuahigginsonCH — APPROVED** [review]
